### ISSUE TITLE
Change PWM Frequency tp 25kHz?

### ISSUE THREAD

**Issue by Ch3fk0ch88** [sarfata/pi-blaster#107]
Hy, 
on which way i can cahnge the PWM Frequency to 25kHz?
I found nothing in the internet....

Thanks

**Comment by sarfata** [comment]
The README gives some explanation: https://practicingruby.com/articles/tricks-for-working-with-text-and-files but it seems a bit outdated.

For 25kHz, you will need a period of 40us. If you change [`CYCLE_TIME_US `](https://github.com/sarfata/pi-blaster/blob/7af3d142573dc177c285c5c56d5f6f426d22a09c/pi-blaster.c#L111) in the source code to 4 instead of 1000 you should get a 25kHz PWM. Be aware however that you will only have 4 steps to adjust the duty rate of the PWM, roughly: 25%, 50%, 75%, 100%.

To improve this, you can reduce `SAMPLE_US` and set it to 2us (or maybe even 1us) and you will get 40us/2us = 20 (or 40) steps for your PWM.

Let me know how that works for you!

**Comment by sarfata** [comment]
After the changes in the code, the startup code shows:

```
pi@raspberrypi:~/pi-blaster $ sudo ./pi-blaster -D
MBox Board Revision: 0x10
DMA Channels Info: 0x7f35, using DMA Channel: 14
Using hardware:                   PWM
Number of channels:                32
PWM frequency:               25000 Hz
PWM steps:                         20
Maximum period (100  %):         40us
Minimum period (5.000%):          2us
DMA Base:                  0x20007000
Initialised, Reading /dev/pi-blaster.
```

I will try to hook up a oscilloscope and confirm everything still works but it should ;)

**Comment by Ch3fk0ch88** [comment]
So, 
in tried it but i get no PWM Signal^^

**Comment by Ch3fk0ch88** [comment]
So,
i tested it again but i dont get a signal with the changed pi-blaster.c

**Comment by sarfata** [comment]
Yes I have done some tests too and I it does not work when there are less
than 200 samples (cycles period divided by time unit).

So right now the highest frequency available is 1/200us so 5kHz.

I am pretty sure it’s a bug in pi-blaster somewhere but I have not figured
where yet.

**Comment by Ch3fk0ch88** [comment]
Ok,
Let me know when u find a solution.

Thanks

**Comment by Ch3fk0ch88** [comment]
Hello,
It's been a while, have you found the bug by now?

**Comment by sarfata** [comment]
No sorry. I have not been working on this project.

**Comment by madprunes** [comment]
I am also having this issue, although the fans I'm controlling appear to work at 5khz their spec sheet says 21k to 28k. I have also had a play and confirmed the minimum 200 samples is still the issue.

**Comment by sarfata** [comment]
Would be great if someone could do a PR to fix this. I would gladly review and merge it.
